src/transaction/creation_common.rs

```diff
@@ -0,0 +1,883 @@
+        Arg::new(name)
+            .long(name)
+            .required(false)
+            .action(ArgAction::Append)
+            .value_name(ARG_VALUE_NAME)
+            .help(ARG_HELP.as_str())
+            .display_order(order)
+    }
+}
+
+/// Handles providing the arg for and retrieval of complex session and payment args. These are read
+/// in from a file.
+pub(super) mod args_complex {
+    use super::*;
+
+    const ARG_VALUE_NAME: &str = common::ARG_PATH;
+    const ARG_HELP: &str =
+        "Path to file containing 'ToBytes'-encoded named and typed args for passing to the Wasm \
+        code";
+
+    pub(in crate::transaction) mod session {
+        use super::*;
+
+        pub const ARG_NAME: &str = "session-args-complex";
+
+        pub fn arg() -> Arg {
+            super::arg(ARG_NAME, DisplayOrder::SessionArgsComplex as usize)
+                .requires(super::session::ARG_NAME)
+        }
+
+        pub fn get(matches: &ArgMatches) -> &str {
+            matches
+                .get_one::<String>(ARG_NAME)
+                .map(String::as_str)
+                .unwrap_or_default()
+        }
+    }
+
+    fn arg(name: &'static str, order: usize) -> Arg {
+        Arg::new(name)
+            .long(name)
+            .required(false)
+            .value_name(ARG_VALUE_NAME)
+            .help(ARG_HELP)
+            .display_order(order)
+    }
+}
+
+pub(super) mod payment_amount{
+    use super::*;
+    pub(in crate::transaction) const ARG_NAME: &str = "payment-amount";
+
+    const ARG_VALUE_NAME: &str = common::ARG_INTEGER;
+
+    const ARG_SHORT: char = 'p';
+    const ARG_HELP: &str = "Uses the standard-payment system contract. The value is the amount arg \
+                            of the standard-payment contract";
+
+    pub(in crate::transaction) fn arg() -> Arg {
+        Arg::new(ARG_NAME)
+            .long(ARG_NAME)
+            .short(ARG_SHORT)
+            .required(false)
+            .value_name(ARG_VALUE_NAME)
+            .help(ARG_HELP)
+            .display_order(DisplayOrder::PaymentAmmount as usize)
+    }
+
+    pub fn get(matches: &ArgMatches) -> Option<&str> {
+        matches.get_one::<String>(ARG_NAME).map(String::as_str)
+    }
+}
+
+pub(super) mod pricing_mode {
+    use super::*;
+    pub(in crate::transaction) const ARG_NAME: &str = "pricing-mode";
+
+    const ARG_VALUE_NAME: &str = common::ARG_STRING;
+    const ARG_HELP: &str = "Used to identify the payment mode chosen to execute the transaction";
+
+    pub(in crate::transaction) fn arg() -> Arg {
+        Arg::new(ARG_NAME)
+            .long(ARG_NAME)
+            .required(false)
+            .value_name(ARG_VALUE_NAME)
+            .help(ARG_HELP)
+            .display_order(DisplayOrder::PaymentMode as usize)
+    }
+
+    pub fn get(matches: &ArgMatches) -> Option<&str> {
+        matches.get_one::<String>(payment_amount::ARG_NAME).map(String::as_str)
+    }
+}
+
+pub(super) mod initiator_address {
```

Comment:
This module should really be merged with the `session_account` one below as `initiator_address` is its replacement.  We eventually want the same functionality in terms if allowing the user to specify a public key file path (hex or pem), a formatted public key, a formatted account hash, a formatted entity address, or nothing (in which case the public key associated with the secret key is used).